### PR TITLE
[FIRE-44] State builder UI

### PR DESCRIPTION
**Pull request by Kamil153**

### CODE REVIEW

**Comment by koszar91** [comment]
Conflicts!

**Comment by koszar91** [comment]
Conflicts!

**Comment by koszar91** [comment]
My rooms are always failing to load.
I build simple two rooms with a door (air gap) on their common wall. Then I hit save, and then try to run simulation or edit the room, and it's not working, and in the logs I see: `2022-11-13 11:59:25 ERROR RoomStorage : Failed to load room "tomek2"`

**Comment by koszar91** [comment]
Bug 2
When I don't give a name to the scene (or first element in the scene, I'm not sure) the app crashes with this error log:
```
Dear ImGui Assertion Failed: id != window->ID && "Cannot have an empty ID at the root of a window. If you need an empty label, use ## and read the FAQ about how the ID Stack works!"
Assertion Located At: /tmp/imgui/jni/imgui.cpp:8412
java.lang.Exception: Stack trace
	at java.base/java.lang.Thread.dumpStack(Thread.java:1380)
	at imgui.ImGui$1.imAssertCallback(ImGui.java:85)
	at imgui.assertion.ImAssertCallback.imAssert(ImAssertCallback.java:21)
	at imgui.ImGui.selectable(Native Method)
	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.renderEditMenu(StateBuilderScene.java:90)
	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.renderGUI(StateBuilderScene.java:67)
	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.update(StateBuilderScene.java:62)
	at pl.edu.agh.firecell.core.Window.run(Window.java:87)
	at pl.edu.agh.firecell.FirecellApplication.main(FirecellApplication.java:17)
```

**Comment by koszar91** [comment]
Issue 1
I cannot ignite anything, I can't find this option in editor.

**Comment by koszar91** [comment]
Issue 2
The ImGUI windows are kind of hard to read when you start the application:
image
Idk what's going on here. Please reorganize them.

**Comment by koszar91** [comment]
Issue 3
It would be great if you delete the windows at all :P
And move the whole UI to the top bar, for the user to click on option in this top bar, and the sub-menu appears, for example form for adding elements.

**Review by koszar91 — CHANGES_REQUESTED** [review]
Functionality looks really great 💪 💪 💪  
But we still need some polishing to the UI and bugfixing :c

**Comment by Kamil153** [comment]
> Bug 2 When I don't give a name to the scene (or first element in the scene, I'm not sure) and try to save or add element, the app crashes with this error log:
> 
> ```
> Dear ImGui Assertion Failed: id != window->ID && "Cannot have an empty ID at the root of a window. If you need an empty label, use ## and read the FAQ about how the ID Stack works!"
> Assertion Located At: /tmp/imgui/jni/imgui.cpp:8412
> java.lang.Exception: Stack trace
> 	at java.base/java.lang.Thread.dumpStack(Thread.java:1380)
> 	at imgui.ImGui$1.imAssertCallback(ImGui.java:85)
> 	at imgui.assertion.ImAssertCallback.imAssert(ImAssertCallback.java:21)
> 	at imgui.ImGui.selectable(Native Method)
> 	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.renderEditMenu(StateBuilderScene.java:90)
> 	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.renderGUI(StateBuilderScene.java:67)
> 	at pl.edu.agh.firecell.core.statebuilder.StateBuilderScene.update(StateBuilderScene.java:62)
> 	at pl.edu.agh.firecell.core.Window.run(Window.java:87)
> 	at pl.edu.agh.firecell.FirecellApplication.main(FirecellApplication.java:17)
> ```

I've created separate story for input validation. I've added default room and element names for now.

File: src/main/java/pl/edu/agh/firecell/core/statebuilder/element/IgnitionElement.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public record IgnitionElement(Vector3i position, Vector3i size) implements BaseCuboidElement {
+
+    @Override
+    public void addTo(List<Cell> cells, Vector3i spaceSize) {
+        IndexUtils.range(position, new Vector3i(-1).add(position).add(size))
+                .map(expandedIndex -> IndexUtils.flattenIndex(expandedIndex, spaceSize))
+                .filter(index -> cells.get(index).flammable())
+                .forEach(index -> {
+                    var oldCell = cells.get(index);
+                    cells.set(index, new Cell(400, 1, oldCell.flammable(),
```
Comment (koszar91) [inline src/main/java/pl/edu/agh/firecell/core/statebuilder/element/IgnitionElement.java:18]: Here you set an arbitrary start temperature. That's wrong, take it from the material of the cell.

**Review by koszar91 — CHANGES_REQUESTED** [review]
Absolute perfection. Good job! Just fix one thing and resolve conflicts with develop (if needed) and we can merge

**Review by koszar91 — APPROVED** [review]
GG 💪

**Pull request merged**